src/HealthChecks/Abstractions/src/HealthReportEntry.cs

```diff
@@ -22,13 +23,15 @@ public struct HealthReportEntry
         /// <param name="duration">A value indicating the health execution duration.</param>
         /// <param name="exception">An <see cref="Exception"/> representing the exception that was thrown when checking for status (if any).</param>
         /// <param name="data">Additional key-value pairs describing the health of the component.</param>
-        public HealthReportEntry(HealthStatus status, string description, TimeSpan duration, Exception exception, IReadOnlyDictionary<string, object> data)
+        /// <param name="tags">Tags associated with the health check that generated the report entry.</param>
+        public HealthReportEntry(HealthStatus status, string description, TimeSpan duration, Exception exception, IReadOnlyDictionary<string, object> data, IEnumerable<string> tags = null)
```

Comment:
We can't change an existing constructor like this as it's a binary breaking change.

Add a new constructor with the extra parameter and update the old one to call the new one.